### PR TITLE
Add to_equation for SPMe

### PR DESCRIPTION
**Pull request by priyanshuone6**
# Description

- Add to_equation in Broadcasts, Concatenation, Log, Sinh, Integral and Independent Variable
- Modify print_name

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/expression_tree/concatenations.py [hunk 1]
```diff
@@ -114,6 +115,25 @@ def is_constant(self):
         """See :meth:`pybamm.Symbol.is_constant()`."""
         return all(child.is_constant() for child in self.children)
 
+    def _sympy_operator(self, *children):
+        """Apply appropriate SymPy operators."""
+        concat_str = r"\\".join(map(sympy.latex, children))
+        concat_sym = sympy.Symbol(r"\begin{cases}" + concat_str + r"\end{cases}")
```
Comment (tinosulzer) [inline pybamm/expression_tree/concatenations.py:121]: these should also have indications of what domain each case holds in. Can be a separate PR
Screen Shot 2021-07-13 at 1 15 03 PM

File: pybamm/expression_tree/unary_operators.py [hunk 2]
```diff
@@ -618,6 +618,10 @@ def _evaluates_on_edges(self, dimension):
         """See :meth:`pybamm.Symbol._evaluates_on_edges()`."""
         return False
 
+    def _sympy_operator(self, child):
+        """Override :meth:`pybamm.UnaryOperator._sympy_operator`"""
+        return sympy.Integral(child, self.integration_variable)
```
Comment (tinosulzer) [inline pybamm/expression_tree/unary_operators.py:623]: can sympy integrals take integral limits? you can get these from the geometry. can be a separate PR

File: pybamm/models/submodels/electrolyte_conductivity/composite_conductivity.py [hunk 3]
```diff
@@ -153,4 +153,7 @@ def get_coupled_variables(self, variables):
         variables.update(self._get_standard_current_variables(i_e))
         variables.update(self._get_split_overpotential(eta_c_av, delta_phi_e_av))
 
+        # Override print_name
+        i_e.print_name = "I_e"
```
Comment (tinosulzer) [inline pybamm/models/submodels/electrolyte_conductivity/composite_conductivity.py:157]: should be lowercase i

File: pybamm/models/submodels/electrolyte_diffusion/full_diffusion.py [hunk 4]
```diff
@@ -82,6 +82,7 @@ def set_rhs(self, variables):
         div_Vbox = variables["Transverse volume-averaged acceleration"]
 
         sum_s_j = variables["Sum of electrolyte reaction source terms"]
+        sum_s_j.print_name = "A"
```
Comment (tinosulzer) [inline pybamm/models/submodels/electrolyte_diffusion/full_diffusion.py:85]: lowercase a

File: pybamm/models/submodels/interface/base_interface.py [hunk 5]
```diff
@@ -353,6 +353,10 @@ def _get_standard_whole_cell_interfacial_current_variables(self, variables):
             a_n, pybamm.FullBroadcast(0, "separator", "current collector"), a_p
         )
 
+        # Override print_name
+        j.print_name = "J"
+        a.print_name = "A"
```
Comment (tinosulzer) [inline pybamm/models/submodels/interface/base_interface.py:358]: lowercase a

File: pybamm/expression_tree/functions.py [hunk 6]
```diff
@@ -446,6 +450,10 @@ def _function_diff(self, children, idx):
         """See :meth:`pybamm.Function._function_diff()`."""
         return Cosh(children[0])
 
+    def _sympy_operator(self, child):
+        """Override :meth:`pybamm.Function._sympy_operator`"""
+        return sympy.sinh(child)
```
Comment (tinosulzer) [inline pybamm/expression_tree/functions.py:455]: could have a default behavior (e.g. in SpecificFunction._sympy_operator) for functions that have the same name in pybamm and sympy:
```python3
class_name = self.__class__.__name__.lower()
sympy_function = getattr(sympy, class_name)
return sympy_function(child)
```

**Review by tinosulzer — COMMENTED** [review]
Looks good, some suggestions

File: pybamm/expression_tree/unary_operators.py (hunk 2 above)
Comment (priyanshuone6) [reply pybamm/expression_tree/unary_operators.py:623]: How to access model.default_geometry from here?

Comment (brosaplanella) [reply pybamm/expression_tree/unary_operators.py:623]: Does the `integration_variable` have this information? Otherwise might be a bit tricky...

Comment (tinosulzer) [reply pybamm/expression_tree/unary_operators.py:623]: Hmm good point, you can't access default geometry from here, so don't worry about it. Maybe instead we could print `\Omega_n` (/s/p) as the lower limit to indicate integration over the entire domain?

Comment (priyanshuone6) [reply pybamm/expression_tree/unary_operators.py:623]: `sympy.Integral(child, sympy.symbols("\Omega_n"))` will return this

Capture

Comment (tinosulzer) [reply pybamm/expression_tree/unary_operators.py:623]: Ok leave it as xn for now

File: pybamm/models/submodels/tortuosity/base_tortuosity.py [hunk 7]
```diff
@@ -51,4 +51,7 @@ def _get_standard_tortuosity_variables(
             }
             variables.update(leading_order_variables)
 
+        # Override print_name
+        tor.print_name = r"\epsilon^{be}"
```
Comment (brosaplanella) [inline pybamm/models/submodels/tortuosity/base_tortuosity.py:55]:
```suggestion
        tor.print_name = r"\epsilon^{b_e}"
```
I think we want the exponent to be `b_e`, can you confirm @tinosulzer?

**Review by brosaplanella — APPROVED** [review]
Just a small comment, the rest looks good. Thanks Priyanshu!

Also, couldn't find what's the issue with the coverage so it should be fine to merge the PR anyway.
